search/lib/api/wfs.js

```diff
@@ -143,8 +207,38 @@ async function getGranules (request, response) {
     }
     // convert STAC params to CMR Params
     const cmrParams = await cmr.convertParams(providerId, params);
-    const granulesResult = await cmr.findGranules(cmrParams);
-
+  
+    let granulesResult; 
+    if ( settings.cmrStacRelativeRootUrl === "/cloudstac") {
+      //Preserve collection_concept_id and concept_id in cmrParams before deleting.
+      //After checking collection_concept_ids being cloud holding collections, they
+      //will be added back one by one because of POST search request requirement.
+      const collection_concept_ids = cmrParams.collection_concept_id;
+      const concept_ids = cmrParams.collection_concept_id;
+      delete cmrParams.collection_concept_id; 
+      delete cmrParams.concept_id;
+
+      //Find all the cloud holding collections applicable
+      //i.e. if collection_concept_ids are present, we will get all the cloud holding collections within these ids.
+      //otherwise, we will get all the cloud holding collections for the provider.
+      const allCloudCollections = await findCloudCollections(providerId, collection_concept_ids);
+      const postSearchParams = new URLSearchParams(cmrParams);
+      if (allCloudCollections.length !== 0) {
+        for (i = 0; i < allCloudCollections.length; i++) {
+          postSearchParams.append("collection_concept_id", allCloudCollections[i]);
+        }
+      } else {
+        return response.status(400).json(`Cloud holding collections not found for provider [${providerId}].`);
+      }
+      if (concept_ids) {
+        for (i = 0; i < concept_ids.length; i++) {
```

Comment:
style change, prefer `concept_ids.forEach(` over `for`

Reply:
Done.